### PR TITLE
MiscUtils word_wrap: Add option to trim only leading whitespace after wrapping

### PR DESCRIPTION
**Pull request by Bumber64**
Replace ``bool collapse_whitespace`` option with an enum. Options ``WSMODE_KEEP_ALL``, ``WSMODE_COLLAPSE_ALL``, and ``WSMODE_TRIM_LEADING``.

Related to https://github.com/DFHack/dfhack/pull/2113

### CODE REVIEW

**Comment by Bumber64** [comment]
``trim_whitespace_after_newline`` is a bit wordy. Any better suggestions?

File: docs/changelog.txt [hunk 1]
```diff
@@ -47,6 +47,7 @@ changelog.txt uses a syntax similar to RST, with a few special sequences:
 ## Documentation
 
 ## API
+- `word_wrap`: add ``trim_whitespace_after_newline`` option, which trims any whitespace at the beginning of a line after wrapping
```
Comment (Bumber64) [inline docs/changelog.txt:50]: This causes ``docs/_auto/news.rst:29: WARNING: undefined label: word_wrap``

Comment (myk002) [reply docs/changelog.txt:50]: yeah, it's not a document label. I'd suggest double backticks.

File: library/MiscUtils.cpp [hunk 2]
```diff
@@ -169,14 +169,15 @@ std::string to_search_normalized(const std::string &str)
 }
 
 
-bool word_wrap(std::vector<std::string> *out, const std::string &str,
-               size_t line_length, bool collapse_whitespace)
+bool word_wrap(std::vector<std::string> *out, const std::string &str, size_t line_length,
+               bool collapse_whitespace, bool trim_whitespace_after_newline)
```
Comment (myk002) [inline library/MiscUtils.cpp:173]: since this is mutually exclusive with the collapse_whitespace boolean, maybe it would be better expressed as an enum?

Comment (Bumber64) [reply library/MiscUtils.cpp]: I made an attempt. Let me know what I need to fix.

**Review by myk002 — APPROVED** [review]

File: library/MiscUtils.cpp (hunk 2 above)
Comment (myk002) [reply library/MiscUtils.cpp]: `word_wrap` itself is not in a namespace, so I think it's fine how you have it. The name of the enum and its values are unique enough that I don't think there is much danger of collision.

File: library/include/MiscUtils.h [hunk 3]
```diff
@@ -389,10 +389,16 @@ DFHACK_EXPORT std::string toUpper(const std::string &str);
 DFHACK_EXPORT std::string toLower(const std::string &str);
 DFHACK_EXPORT std::string to_search_normalized(const std::string &str);
 
+enum word_wrap_wspace_mode {
```
Comment (lethosor) [inline library/include/MiscUtils.h:392]: I suggest `whitespace` over `wspace`, but LGTM either way

**Review by lethosor — APPROVED** [review]
